migrations/2024-05-05-162540_add_image_detail_table/up.sql

```diff
@@ -0,0 +1,16 @@
+-- Drop the id column from the remote_image table, just use link
+ALTER TABLE remote_image
+    DROP COLUMN id,
+    ADD PRIMARY KEY (link),
+    DROP CONSTRAINT remote_image_link_key;
+
+-- No good way to do references here unfortunately, unless we combine the images tables
+-- The link should be the URL, not the pictrs_alias, to allow joining from post.thumbnail_url
```

Comment:
`local_image` uses the pictrs_alias, whereas `remote_image` uses a full url. Because of that there's no way to create a combined images table, and do a foreign key reference to it.

Reply:
You can do it the other way round, add an `id` field in `image_details` and reference that in `remote_image` and `local_image`. Better than storing the link twice.

Reply:
I'll open up an issue for this, but in the future I recommend we consolidate these tables, and add a `local` column.

> You can do it the other way round, add an id field in image_details and reference that in remote_image and local_image.

We can't do that because image_details are optional, and we couldn't generate them for all historical pictures either.

Reply:
So make the foreign key nullable, I dont see the problem with that.

Reply:
I don't agree with that, details can be missing, and they're entirely optional, so it doesn't make sense to use that as the primary table.

In the future I'll probably do a single image table, with a local column, but even then image_details will still be an external optional table join.

Reply:
It wouldnt be a primary table, simply something like this:
```
ALTER TABLE remote_image
  ADD COLUMN image_details_id integer
```

Which means image details are still optional.

Reply:
That's not usually the right way to link data with SQL.

In the future, I'll have a unified `image` table, with an `id` and a `local: boolean` column, and `image_details` (which is optional), will link to it with an `image_id` column. 

That can be done in a later PR tho.